Spigot-Server-Patches/0581-Lazily-track-plugin-scoreboards-by-default.patch

```diff
@@ -0,0 +1,102 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Andrew Steinborn <git@steinborn.me>
+Date: Sat, 3 Oct 2020 04:15:09 -0400
+Subject: [PATCH] Lazily track plugin scoreboards by default
+
+On servers with plugins that constantly churn through scoreboards, there is a risk of
+degraded GC performance due to the number of scoreboards held on by weak references.
+Most plugins don't even need the (vanilla) functionality that requires all plugin
+scoreboards to be tracked by the server. Instead, only track scoreboards when an
+objective is added with a non-dummy criteria.
+
+This is a breaking change, however the change is a much more sensible default. In case
+this breaks your workflow you can always force all scoreboards to be tracked with
+settings.always-track-plugin-scoreboards in paper.yml.
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index f65d3545039a2c471819ab5950c171ef6193cdb4..9fae20c5c1344d905cb2a20fab473af685a01a55 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -457,4 +457,9 @@ public class PaperConfig {
+     private static void maxJoinsPerTick() {
+         maxJoinsPerTick = getInt("settings.max-joins-per-tick", 3);
+     }
++
++    public static boolean alwaysTrackPluginScoreboards;
++    private static void alwaysTrackPluginScoreboards() {
++        alwaysTrackPluginScoreboards = getBoolean("settings.always-track-plugin-scoreboards", false);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/ScoreboardServer.java b/src/main/java/net/minecraft/server/ScoreboardServer.java
+index 929397a4ca8af8aba24c36bd6fc8ed0071982b64..b7605d2850d440cd0c4924d9e430c590e0683f98 100644
+--- a/src/main/java/net/minecraft/server/ScoreboardServer.java
++++ b/src/main/java/net/minecraft/server/ScoreboardServer.java
+@@ -10,7 +10,7 @@ import javax.annotation.Nullable;
+
+ public class ScoreboardServer extends Scoreboard {
+
+-    private final MinecraftServer a;
++    private final MinecraftServer a; public final MinecraftServer getServer() { return this.a; } // Paper - OBFHELPER
```

Comment:
Can drop this and just use minecraftserver getserver instead, don't need a new diff when one exists to get the server already